app/src/main/java/fr/neamar/kiss/result/AppResult.java

```diff
@@ -109,11 +110,31 @@ protected Boolean popupMenuClickHandler(Context context, RecordAdapter parent, M
             case R.id.item_app_hibernate:
                 hibernate(context, appPojo);
                 return true;
+            case R.id.item_exclude:
+                // remove item since it will be hiddden
+                parent.removeResult(this);
+                excludeFromSettings(context, appPojo);
+                return true;
         }
 
         return super.popupMenuClickHandler(context, parent, item);
     }
 
+    private void excludeFromSettings(Context context, AppPojo appPojo)
+    {
+
+        String excludedAppList = PreferenceManager.getDefaultSharedPreferences(context).
+                getString("excluded-apps-list", context.getPackageName() + ";");
+        PreferenceManager.getDefaultSharedPreferences(context).edit()
+                .putString("excluded-apps-list", excludedAppList + appPojo.packageName + ";").commit();
+        //remove app pojo from appProvider results - no need to reset handler
+        KissApplication.getDataHandler(context).getAppProvider().removeApp(appPojo);
+        if(context instanceof MainActivity) {
+            ((MainActivity) context).retrieveFavorites();
```

Comment:
curious here. You're doing this to ensure that the favorite list is up to date, for instance in the use case where the user is currently displaying the KISS bar, and removing an item currently in the list, right?